txn: fix the maximum value of a transaction from 10GB to 1 TiB (#19091)
pingcap/docs: develop/dev-guide-transaction-restraints.md

The basic principle is to limit the size of the transaction. At the KV level, TiDB has a restriction on the size of a single transaction. At the SQL level, one row of data is mapped to one KV entry, and each additional index will add one KV entry. The restriction is as follows at the SQL level:

- The maximum single row record size is 120 MiB.

    - You can adjust it by using the [`performance.txn-entry-size-limit`](/tidb-configuration-file.md#txn-entry-size-limit-new-in-v4010-and-v500) configuration parameter of tidb-server for TiDB v4.0.10 and later v4.0.x versions, TiDB v5.0.0 and later versions. The value is `6 MB` for versions earlier than v4.0.10.
    - Starting from v7.6.0, you can use the [`tidb_txn_entry_size_limit`](/system-variables.md#tidb_txn_entry_size_limit-introduced-from-v760-version) system variable to dynamically modify the value of this configuration item.

- The maximum single transaction size supported is 1 TiB. 

>     - You can configure it by [`performance.txn-total-size-limit`](/tidb-configuration-file.md#txn-total-size-limit) for TiDB v4.0 and later versions. The value is `100 MB` for earlier versions.

Reviewer: ```suggestion
    - For TiDB v4.0 and later versions, You can configure it by [`performance.txn-total-size-limit`](/tidb-configuration-file.md#txn-total-size-limit). The value is `100 MB` for earlier versions.
```

>     - For TiDB v4.0 and later versions, You can configure it by [`performance.txn-total-size-limit`](/tidb-configuration-file.md#txn-total-size-limit). The value is `100 MB` for earlier versions.
>     - For TiDB v6.5.0 and later versions, this configuration is no longer recommended. For more information, see [`performance.txn-total-size-limit`](/tidb-configuration-file.md#txn-total-size-limit).

Author: ```suggestion
    - For TiDB v4.0 and later versions, You can configure it by [`performance.txn-total-size-limit`](https://docs.pingcap.com/tidb/stable/tidb-configuration-file#txn-total-size-limit)). The value is `100 MB` for earlier versions.
    - For TiDB v6.5.0 and later versions, this configuration is no longer recommended. For more information, see [`performance.txn-total-size-limit`](https://docs.pingcap.com/tidb/stable/tidb-configuration-file#txn-total-size-limit)).
```

>     - In TiDB v6.5.0 and later versions, this configuration is no longer recommended. For more information, see [`performance.txn-total-size-limit`](/tidb-configuration-file.md#txn-total-size-limit).

Author: ```suggestion
    - For TiDB v6.5.0 and later versions, this configuration is no longer recommended. For more information, see [`performance.txn-total-size-limit`](/tidb-configuration-file.md#txn-total-size-limit).
```
